cmake/FetchBoost.cmake

```diff
@@ -0,0 +1,33 @@
+include(FetchContent)
+
+message(STATUS "Fetching Boost headers from source...")
```

Comment:
I am not sure about this. I'd rather remove boost altogether.
Dependencies are so easy on a normal linux distro, the rest is a just a jungle.